tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/GitHubService.cs

```diff
@@ -311,5 +321,83 @@ public async Task<Issue> GetIssueAsync(string repoOwner, string repoName, int is
                 throw;
             }
         }
+
+        public async Task UpdateFileAsync(string owner, string repoName, string path, string message, string content, string sha, string branch)
+        {
+            try
+            {
+                var updateRequest = new UpdateFileRequest(message, content, sha, branch);
+                var result = await gitHubClient.Repository.Content.UpdateFile(owner, repoName, path, updateRequest);
+            }
+            catch (NotFoundException ex)
+            {
+                logger.LogInformation("GitHubService: Path {path} not found in {owner}/{repoName} on reference {branch}. Exception: {exception}", path, owner, repoName, branch, ex.Message);
+                throw;
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "GitHubService: Error fetching contents from {owner}/{repoName}/{path} on reference {branch}", owner, repoName, path, branch);
+                throw;
+            }
+        }
+
+        public async Task<CreateBranchStatus> CreateBranchAsync(string repoOwner, string repoName, string branchName, string baseBranchName = "main")
+        {
+            try
+            {
+                var branchExists = await GetBranchAsync(repoOwner, repoName, branchName);
+                if (branchExists)
+                {
+                    return CreateBranchStatus.AlreadyExists;
+                }
+
+                // Get the base branch reference first
+                var baseReference = await gitHubClient.Git.Reference.Get(repoOwner, repoName, $"heads/{baseBranchName}");
+
+                // Create the new branch reference
+                var newReference = new NewReference("refs/heads/" + branchName, baseReference.Object.Sha);
+                var createdReference = await gitHubClient.Git.Reference.Create(repoOwner, repoName, newReference);
+
+                return CreateBranchStatus.Created;
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, $"Failed to create branch {branchName} in {repoOwner}/{repoName}");
+                throw;
+            }
+        }
+
+        public async Task<bool> GetBranchAsync(string repoOwner, string repoName, string branchName)
+        {
+            try
+            {
+                var branch = await gitHubClient.Repository.Branch.Get(repoOwner, repoName, branchName);
+                return branch != null;
+            }
+            catch (NotFoundException)
+            {
+                return false;
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Error getting branch '{branchName}' in {repoOwner}/{repoName}: {ex.Message}");
+                return false;
+            }
+        }
+
+        public async Task<(IReadOnlyList<RepositoryContent>, string)> GetFileContentsAsync()
```

Comment:
This should still take org, repo and file path as parameters since it's a generic helper function that anyone could use.

Reply:
Also we can just return `string` only with the contents. If the caller needs the RepositoryContent list then they can just call `GetContentsAsync`.